### DISCUSSION TITLE
USB connection howto

### DISCUSSION THREAD

**Discussion by berseker** [BimmerGestalt/AAIdrive#628] [Q&A]
Dear All, I struggle to have a "fluid" experience with Mapbox\Screenmirroring, and I think my problem is related to usb connection. is there some "walkthrough" to use USB connection properly? do i have to disable bluetooth to use it? I would like to use if possible both connection simultaneusly

- Bluetooth for music
- USB for the higher Mapbox framerate

**Comment by hufman** [comment]
As I understand it, the car only allows the Apps protocol over a single connection at a time. When you enter the Mobile Devices menu for your phone's Bluetooth or USB connection and click Apps, it will switch the protocol to that connection and reject attempts on the other connection.
You don't have to disable Bluetooth, and playing music over Bluetooth works fine with the USB apps protocol. The tricky part is that MyBMW may be trying to connect to the wrong connection, and may not know to switch if you switch the car's connection.
I believe that, if Bluetooth Apps were running, the car will default the USB to only be storage, even if you later plug in without Bluetooth connected.
That's about as much as I know!

**Reply by berseker** [reply]
thank you for the explanation. I will do some experiments & revert if something is wrong! thank you

**Comment by DrTermi** [comment]
Indeed using apps using the USB is a lot more smooth. The screen mirroring for me doesnt work more than 30 seconds using the Bluetooth connection (idk why) but using the usb the connection stays as long as i want. the only problem is i didnt manage yet to get sound from bluetooth and apps through USB at the same time yet (maybe the Bluetooth will connect also when the engine is on as i havent tried when the engine is on yet). 
to properly connect what i do is just to connect the usb to my phone and put it to tranfer data mode. Phone asks me which app i want to use (i have MyBMW and BMW Connected both) so i choose myBMW then it connects to the car. Then i open aaidrive and it's automatically connected to the car. The apps are added a few seconds later in connected drive of the car.
